pyam/core.py

```diff
@@ -1358,6 +1388,10 @@ def _check_rows(rows, check, in_range=True, return_test='any'):
         msg = 'Unknown checking type: {}'
         raise ValueError(msg.format(check.keys() - valid_checks))
 
+    rows = rows.copy()
```

Comment:
should this copy move into the `if`statement? otherwise we might be doing a copy unnecessarily?

Reply:
It's to ensure you don't accidentally add  a `year` column to your dataframe

Reply:
But wouldn't that only happen if you get inside the `if` statement? Suggestion would be:

```
    if 'year' not in rows:
        rows = rows.copy()
        rows['year'] = rows['time'].apply(lambda x: x.year)
```

Reply:
Or perhaps I am confused. Sorry if that's the case!

Reply:
ah yep I get it, nice!